pkg/targetgroupbinding/resource_manager.go

```diff
@@ -461,3 +475,26 @@ func isELBV2TargetGroupNotFoundError(err error) bool {
 	}
 	return false
 }
+
+func isELBV2TargetInELBVPC(podIP string, vpc *ec2sdk.Vpc) bool {
+	// Check if the pod IP is in the primary VPC CIDR.
+	if isIPinCIDR(podIP, *vpc.CidrBlock) {
+		return true
+	}
+
+	// Check if the pod IP is from a secondary CIDR block.
+	for _, v := range vpc.CidrBlockAssociationSet {
```

Comment:
1. CidrBlockAssociationSet is not "secondary CIDR block", it contains the primary CIDR as well, so we don't need to check the primary CIDR above.
2. let's always use awssdk.StringValue(v.CidrBlock) when dereference strings from aws sdk